### PR TITLE
added features for better calculation precision

### PR DESCRIPTION
**Pull request by Storas008**
The algorithm is in working progress and can be improved.

### CODE REVIEW

File: Services/EnergyCostAlgorithm.cs [hunk 1]
```diff
@@ -3,23 +3,32 @@ namespace Ultra_Saver;
...
-    public double totalEnergyCost { get; set;} = 0;
-    private float costPerkWh = 0;
+    public float TotalEnergy { get; set; } = 0;
 
     // powerScale is a proportion of power used for the specified time; scales 1 to 10; ex.: medium heat, 3/10 of power
-    private void ElectricPower (int appliancePower, short powerScale, int timeInMin){
-        totalEnergy += appliancePower * powerScale * timeInMin / 600_000;
-        totalEnergyCost = Math.Round (totalEnergy * costPerkWh, 2);
+    // if User's appliance power efficiancy is 80% then ApplianceEfficiency = 0.8
+    public void ElectricPower (short AppliancePower, byte PowerScale, short TimeInMin, float ApplianceEfficiency){
```
Comment (onlyidev) [inline Services/EnergyCostAlgorithm.cs:10]: There is no need to optimize memory usage to such a level. Moreover, albeit not very likely, it is not completely unreasonable for appliancePower to be > ~32kWh

File: Services/EnergyCostAlgorithm.cs [hunk 2]
```diff
@@ -3,23 +3,32 @@ namespace Ultra_Saver;
...
+        TotalEnergy += AppliancePower * PowerScale * TimeInMin / 600_000 / ApplianceEfficiency;
     }
 
-    private double GasPower (int appliancePower, short powerScale, int timeInMin){
-        return 0;
+    public void GasPower (short AppliancePower, byte PowerScale, short TimeInMin, float ApplianceEfficiency){
+        // small stove - ~1200w , medium - ~1500 - 1800, large - up to 3k
+        // 1 kWh = 0.0947 m3 gas (100% efficient combustion)
+        //TotalEnergy +=  / ApplianceEfficiency
```
Comment (onlyidev) [inline Services/EnergyCostAlgorithm.cs:17]: Excessive commentary. Please add the algorithm as well.

File: Services/EnergyCostAlgorithm.cs [hunk 3]
```diff
@@ -3,23 +3,32 @@ namespace Ultra_Saver;
...
+    public void HeatingPan (short AppliancePower, float ApplianceEfficiency){
+        // considering heating pan is for approximately 2 minutes
+        TotalEnergy += AppliancePower / 30 / ApplianceEfficiency;
+    }
+}
+
+// Research clearly shows that induction cooktops are more energy efficient:
+// gas cooktops are about 40 percent efficient;
+// electric-coil and standard smooth-top electric cooktops are about 74 percent efficient,
+// and induction cooktops are 84 percent efficient
```
Comment (onlyidev) [inline Services/EnergyCostAlgorithm.cs:34]: Excessive commentary. Please remove or condense

File: Services/EnergyCostAlgorithm.cs (hunk 1 above)
Comment (Storas008) [reply Services/EnergyCostAlgorithm.cs]: In my point of view, optimizing memory is always great. Things such as AppliancePower could be declared as int, however PowerScale, which value will always be 1-10, can safely be declared as byte type.

**Review by onlyidev — APPROVED** [review]

**Pull request merged**